services/src/test/java/org/keycloak/vault/VaultTranscriberTest.java

```diff
@@ -0,0 +1,210 @@
+        }
+        // after the try-with-resources block the secret should have been overridden
+        Assert.assertFalse(Arrays.equals("non_vault_secret".getBytes(StandardCharsets.UTF_8), secretBuffer.array()));
+
+        // check that a null expression results in an empty optional instance.
+        try (VaultRawSecret secret = transcriber.getRawSecret(null)) {
+            Assert.assertFalse(secret.getRawSecret().isPresent());
+        }
+    }
+
+    @Test
+    public void testRetrieveCharSecret() throws Exception {
+        CharBuffer secretBuffer;
+
+        // first attempt to obtain a secret using a proper vault string with an existing key.
+        try (VaultCharSecret secret = transcriber.getCharSecret("${vault.vault_key_2}")) {
+            Optional<CharBuffer> optional = secret.getCharSecret();
+            Assert.assertTrue(optional.isPresent());
+            secretBuffer = optional.get();
+            Assert.assertTrue(Arrays.equals("secret2".toCharArray(), secretBuffer.array()));
+        }
+        // after the try-with-resources block the secret should have been overridden
+        Assert.assertFalse(Arrays.equals("secret2".toCharArray(), secretBuffer.array()));
+
+        // now attempt to obtain a secret using a proper vault string with a non-existing key.
+        try (VaultCharSecret secret = transcriber.getCharSecret("${vault.non_existing_key}")) {
+            Optional<CharBuffer> optional = secret.getCharSecret();
+            Assert.assertFalse(optional.isPresent());
+        }
+
+        // finally try to obtain a secret using a string that is not a vault string. The string itself should be encoded.
+        try (VaultCharSecret secret = transcriber.getCharSecret("non_vault_secret")) {
+            Optional<CharBuffer> optional = secret.getCharSecret();
+            Assert.assertTrue(optional.isPresent());
+            secretBuffer = optional.get();
+            Assert.assertTrue(Arrays.equals("non_vault_secret".toCharArray(), secretBuffer.array()));
+        }
+        // after the try-with-resources block the secret should have been overridden
+        Assert.assertFalse(Arrays.equals("non_vault_secret".toCharArray(), secretBuffer.array()));
+
+        // check that a null expression results in an empty optional instance.
+        try (VaultCharSecret secret = transcriber.getCharSecret(null)) {
+            Assert.assertFalse(secret.getCharSecret().isPresent());
+        }
+    }
+
+    @Test
+    public void testRetrieveStringSecret() throws Exception {
+
+        // first attempt to obtain a secret using a proper vault string with an existing key.
+        Optional<WeakReference<String>> optional = transcriber.getStringSecret("${vault.vault_key_1}");
+        Assert.assertTrue(optional.isPresent());
+        WeakReference<String> stringSecret = optional.get();
+        Assert.assertEquals("secret1", stringSecret.get());
```

Comment:
I see, well I understand this is improbable but I would really like if you could add this guard because this might be quite a nasty issue which is hard to debug if this will happen in CI.

Reply:
Leaving this conversation open until we get a decision based on the questions Sebastian asked Andrew / Christine.

Reply:
Ok, so the decision has been to drop the weak references. I'm resolving this conversation based on that.